### ISSUE TITLE
`nonblock-statement-body-position` Errors on Correct Code

### ISSUE THREAD

**Issue by epmatsw** [eslint/eslint#8202]
**Tell us about your environment**

* **ESLint Version:** 3.17.0
* **Node Version:** 7.6.0
* **npm Version:** 4.1.2

**What parser (default, Babel-ESLint, etc.) are you using?**
Default

**Please show your full configuration:**

```
{
    rules: {
        'nonblock-statement-body-position':  2
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js

function blah() {
    if (window.hey) return window.doit({
        asdf: 2,
    });
    return 1;
}
```

**What did you expect to happen?**
No errors.

**What actually happened? Please include the actual, raw output from ESLint.**

`3:21  error  Expected no linebreak before this statement  nonblock-statement-body-position`

Even if the rule doesn't allow expressions that extend onto the next line (which would be a bit of a bummer), the error message doesn't make a whole lot of sense.

**Comment by not-an-aardvark** [comment]
Thanks for the report, I can reproduce this issue.

The intended behavior is that this code should be valid. It looks like the rule is reporting an error when the statement is multiline, even if it starts on the same line as the `if` statement.
